### PR TITLE
[WIP] Migration to yeoman 1.0

### LINKED ISSUE #4631: Prepare for Yeoman 1.0
As mentioned [here](https://github.com/yeoman/generator/issues/766#issuecomment-265658238) Yeoman 1.0 will soon be released.
There is a [blog post](https://github.com/yeoman/yeoman.github.io/blob/a87b3452377b1b232145f910d6a945804bf84d2c/app/_posts/blog/2016-12-03-hello-generator-1.0.md) describing the upgrade process and new features, we should follow it.

### PR DESCRIPTION
**Pull request by deepu105**
I have fixed most issues and it passes `yarn test` now we need to test remaining part to see if there are any more issues

TODO

- [x] Move to ES6 classes where possible
- [ ] Test module hooks as compose syntax have changed
- [x] check performance impact [Workaround implemented]

@jdubois In my local `yarn test` I kind of feel it to be slower than before but It could be coz im on a slower network as im travelling so could you plz check in your machine

Fix #4631

### CODE REVIEW

**Comment by deepu105** [comment]
this is not triggering a travis build which is weird. Any idea @pascalgrimaud

**Comment by pascalgrimaud** [comment]
Build in waiting, it should start soon

About `yarn test`, there is only 3 or 4 tests which are slow because it checks the install. All of the rest are totally offline. The tests concerned by this, don't have `skipChecks: true`

**Comment by jdubois** [comment]
Great!!
Builds are stuck because of the Amazon S3 crash from this evening, they should come back soon

**Comment by deepu105** [comment]
Since there are 2 builds i'm gonna cancel one

File: travis/build-samples.sh [hunk 1]
```diff
@@ -18,34 +18,50 @@ function usage() {
 
 function buildProject() {
     dir=$1
+    JHIPSTER=$dir
```
Comment (deepu105) [inline travis/build-samples.sh]: @pascalgrimaud I modified the travis scripts slightly in this commit. can you take a look?

Comment (pascalgrimaud) [inline travis/build-samples.sh]: This script is not used by Travis at all. It was done by one of our contributors to test locally, before submitting a PR

Very very quick look right now, it seems OK. I will review in details later ;)

Comment (pascalgrimaud) [inline travis/build-samples.sh]: Oh I see, your comment wasn't for this file specially but for all !
I will take a look tonight

**Comment by pascalgrimaud** [comment]
@deepu105 : about the performance impact, I had a very quick look. It comes from the function `writeFilesToDisk` in the `generator-base.js` file.
Nothing has changed but the function takes 5~6 secondes to be executed now. That's why the `yarn test` are very long now

**Comment by deepu105** [comment]
Thanks pascal that was bugging me as well. Ill take a look to see whats
happening

Thanks & regards,
Deepu

**Comment by deepu105** [comment]
@jdubois the tests are green(I cancelled the duplicate build thats why PR build is failed) so do you wanna merge this or do you wanna wait for the performance issue to be addressed?

**Comment by jdubois** [comment]
We could wait 1 or 2 days, but then I would merge this to avoid conflicts

**Comment by deepu105** [comment]
Ok i'm merging this

**Comment by pascalgrimaud** [comment]
very good job @deepu105 !

**Pull request merged**